src/azure/storage/imds_credential.rs

```diff
@@ -0,0 +1,156 @@
+use http::{HeaderValue, Method, Request};
+use reqwest::{Client, Url};
+use serde::Deserialize;
+use std::str;
+
+const MSI_API_VERSION: &str = "2019-08-01";
+const MSI_ENDPOINT: &str = "http://169.254.169.254/metadata/identity/oauth2/token";
+
+/// Attempts authentication using a managed identity that has been assigned to the deployment environment.
+///
+/// This authentication type works in Azure VMs, App Service and Azure Functions applications, as well as the Azure Cloud Shell
+///
+/// Built up from docs at [https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol](https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol)
```

Comment:
We can use `<https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol>`